mantidproject/mantid
# Update Matrix/TableWorkspaceDisplay and InstrumentView on workspace change

**Description of work.**
Adds ADS observers and handles replace/rename/delete/clear events for the Matrix/Table workspace displays, and the instrument view

**To test:**
- Matrix Workspace
  - Load a matrix workspace
  - Show its data
  - Multiply the workspace and see that the numbers are updated
  - Delete the workspace and see that the window is closed
- Table Workspace
  - Do the same workflow as for Matrix
  - A good algorithm to modify the workspace is `SortTableWorkspace` as it will work with any table
- Open the instrument view, and delete the workspace, the window should be closed. This PR does **not** handle replace/rename events for instrument view, as they are handled in C++ `InstrumentWidget`



Fixes #24475 
Fixes #24499

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
